christfellowshipchurch/web-app-v2
# Finder

## About
A first pass integration with new 🛰️ [`search-groups`](https://github.com/christfellowshipchurch/apollos-api/pull/197) API branch, creating the scaffolding for full Finder flow.

Specifically, this PR:

- Creates `useSearchGroups` hook for executing group search
- Creates `GroupFiltersProvider` to manage filters state
- Creates `/community/search` page to execute search and crudely show results
- Allow state handoff between `/community` and `/community/search` pages — serializing and rehydrating filters state via `URLSearchParams`/query string
  - ⚠️ WIP and may be a smarter Next.js way to do this...
- Renames the step components in `GroupFilterModal` to reflect the data in their form

### Caveats/Callouts
- I'm not 100% confident in the technical mechanics of search string serialization and hydration with Next.js — open to feedback!
- Meeting days and Lineups (`subPreferences`) are not integrated to search yet
- You can filter by Crew, Sisterhood, etc (`preference`) but only by going initiating a search from that Community Preference page (no step in the modal)
- Search Results page is WIP UX/design

## Testing

**Note**: You need to use the 🛰️ [`search-groups`](https://github.com/christfellowshipchurch/apollos-api/pull/197) API branch locally.

- Initiate a search from the Community home page
- Initiate a search from a Community Preference page (i.e. Crew), and verify the search is filtered by that `preference`
- Copy/paste a search results into a new tab, and verify state hydrates properly from those initial params
- Change your search criteria from the Search Results page
- etc
